### PR TITLE
Sending and Receiving data from/to the Leaderboard API

### PR DESCRIPTION
**Pull request by SurafelGetachewT**
#In this project, I have modified the Leaderboard project to:

1. Create a new game with the name of your choice by using the API.
2. Implement the "Refresh" button to get all scores for a game created by you from the API.
3. Implement the form "Submit" button to save a score for a game.
4. Use arrow functions instead of the function keyword.
5. Use async and await JavaScript features to consume the API.

### CODE REVIEW

File: src/modules/leaderboard.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+    body: JSON.stringify(scoreObject),
+  });
+};
+
+const submitAddFunc = () => {
```
Comment (danmainah) [inline src/modules/leaderboard.js:57]: - [ ] KIndly check on your form method, when I click submit I get the error below in my browser:
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @SurafelGetachewT, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/modules/leaderboard.js (hunk 1 above)
Comment (SurafelGetachewT) [reply src/modules/leaderboard.js:57]: Hello @danmainah, I have checked in my browser and it works fine

image.

I have added AAA 123 now

File: src/modules/leaderboard.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+const postScore = (scoreObject) => {
+  fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+    body: JSON.stringify(scoreObject),
+  });
+};
+
```
Comment (Munsa1) [inline src/modules/leaderboard.js:56]: - [ ] I am unable to post a score, as I am getting the following error:
image
#### Kindly fix this

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @SurafelGetachewT, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/leaderboard.js (hunk 2 above)
Comment (SurafelGetachewT) [reply src/modules/leaderboard.js:56]: Hello @Munsa1, I have checked it again and it works fine, could you tell me what could be the problem, or could you try it again?

Comment (zainsadaqat) [reply src/modules/leaderboard.js:56]: Can you please record a video submitting a form It's still showing the same issue requested by the previous Code Reviewers.

Comment (SurafelGetachewT) [reply src/modules/leaderboard.js:56]: Please can we have a zoom call to show you please

https://microverse.zoom.us/j/91075584709?pwd=V0RONmZta0VlV2lrbWpFOUxnUmVUUT09

File: src/modules/leaderboard.js (hunk 1 above)
Comment (omarsalem7) [reply src/modules/leaderboard.js:57]: - [ ] As stated by the previous reviewer, as the same problem exists and I stated in [this loom video] (https://www.loom.com/share/e9514ecd04874789ac1defd72de57b27) why this happens if you make sure that your project work on your side kindly make a loom video to demonstrate it is working and the next reviewer will approve it.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @SurafelGetachewT , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Almost near

 ## Highlights
✔️ All linter checks pass
✔️ Descriptive pull request

 ## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @SurafelGetachewT , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No Linter error :+1: 
- Correct GitHub workflow :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**